src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,22 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String sallaryInfo = "";
+        int sallary = 0;
+        dateFrom = dateFrom.trim();
+        dateTo = dateTo.trim();
+
+        for (String name : names) {
+            for (String getData : data) {
+                if (getData.contains(name) && (dateFrom.equals(getData.substring(0,10))
+                            || dateTo.equals(getData.substring(0, 10))
+                            || dateFrom.substring(3, 10).equals(getData.substring(3, 10)))) {
+                    sallary += Integer.parseInt(getData.substring(getData
+                                .lastIndexOf(' ') + 1, getData.length()));
```

Comment:
this is not hardcode and not okay. Use LocalDate, DateTimeFormatter and work with dates, not strings